zokrates_stdlib/stdlib/ecc/edwardsScalarMult.code

```diff
@@ -0,0 +1,26 @@
+import "ecc/edwardsAdd.code" as add
+import "ecc/edwardsOnCurve.code" as assertOnCurve
+// Function that implements scalar multiplication for a given base point
+// Curve parameters are defined with the last argument
+// Note that the exponent array is not check to be boolean in this gadget
+// https://github.com/zcash-hackworks/sapling-crypto/blob/master/src/jubjub/fs.rs#L555
+def main(field[256] exponent, field[2] pt, field[10] context) -> (field[2]):
+
+    field[2] infinity = [context[2], context[3]] 
+
+    field[2] doubledP = pt
+    field[2] accumulatedP = infinity 
+
+    field j = 255 
+    accumulatedP = if exponent[j] == 1 then doubledP else accumulatedP fi
```

Comment:
Is there an issue with making everything fit in the same loop here?

Reply:
one could make it fit into the same loop that way
```
    field[2] doubledP = pt
    field[2] accumulatedP = infinity
    
    for field i in 0..256 do
        field j = 255 - i 
        candidateP = add(accumulatedP, doubledP, context)
        accumulatedP = if exponent[j] == 1 then candidateP else accumulatedP fi
        doubledP = add(doubledP, doubledP, context)
    endfor

    1 == assertOnCurve(accumulatedP, context)
```

def. looks nicer but means we have one additional point doubling that's not needed.